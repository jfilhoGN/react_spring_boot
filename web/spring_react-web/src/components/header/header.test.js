import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './header'

describe('Header', () => {

    //se o componente esta renderizando
    describe('return', () => {
        test('render the component', () => {
            const wrapper = shallow(<Header/>);
            const component = wrapper.dive();
            expect(toJson(component)).toMatchSnapshot();
        });
    });

    describe('NavBar', () => {
        test('click button logout sucessfully', () => {
            const mockOnClick = jest.fn();
            const wrapper = shallow(
                <Header />
            );
            /* const component = wrapper.dive();
            component.find('NavLink').simulate('click');
            expect(mockOnClick.mock.calls.lenght).toEqual(1); */
        });
        
    });
    
});