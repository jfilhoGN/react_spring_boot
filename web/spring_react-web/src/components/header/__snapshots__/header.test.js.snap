// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header return render the component 1`] = `
<div
  className="row"
>
  <Navbar
    className="grey darken-3"
    href="/"
    options={Object {}}
  >
    <div>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          title="login"
          to="/login"
        >
          <Icon>
            assignment_ind
          </Icon>
        </NavLink>
      </li>
    </div>
  </Navbar>
</div>
`;
